File: Angular/src/app/models/product.ts
import { ProductPhoto } from "./productPhoto";

export interface Product{
  id:number;
  productName:string;
  mainPhotoUrl:string;
  productStock:number;
  productPrice:number;
  isActive:boolean;
  categoryName:string;
  brandName:string;
  productPhotos:ProductPhoto[];
}

File: NetCoreApi/Entities/Concrete/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Concrete
{
    public class Product
    {
        public int Id { get; set; }
        public string ProductName { get; set; }
        public int ProductStock { get; set; }
        public decimal ProductPrice { get; set; }
        public bool IsActive { get; set; }
        public int CategoryId { get; set; }
        public int BrandId { get; set; }
        public ICollection<ProductPhoto> ProductPhotos { get; set; }

    }
}
